### PR TITLE
refactor: ORA data / UI layers

### PR DESCRIPTION
**Pull request by nsprenkle**
**TL;DR -** Split of functionality in mixins into data / UI components.

JIRA:
- https://2u-internal.atlassian.net/browse/AU-1398
- https://2u-internal.atlassian.net/browse/AU-1399
- https://2u-internal.atlassian.net/browse/AU-1400
- https://2u-internal.atlassian.net/browse/AU-1401
- https://2u-internal.atlassian.net/browse/AU-1402
- https://2u-internal.atlassian.net/browse/AU-1403
- https://2u-internal.atlassian.net/browse/AU-1404

**What changed?**

- Moved data gathering pieces of mixins into `xblock/apis`
- Created `xblock/ui_mixins/legacy` to replace existing view/handler behaviors from mixins, leveraging our new APIs.
- Moved remaining non-core-xblock functionality from `xblock` to `xblock/utils`.
- Updated references to changed file locations.

**Developer Checklist**

- [x] Reviewed the [release process](https://github.com/openedx/edx-ora2/blob/master/.github/release_process.md)
- [x] Translations and JS/SASS compiled
- [x] Bumped version number in [openassessment/\_\_init\_\_.py](https://github.com/openedx/edx-ora2/blob/master/openassessment/__init__.py#L4) and [package.json](https://github.com/openedx/edx-ora2/blob/master/package.json#L3)

**Testing Instructions**

- [x] Unit tests passing
- [x] Manual tests

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [x] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

File: openassessment/xblock/apis/assessments/self_assessment_api.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from openassessment.assessment.api import self as self_api
+from submissions import api as submission_api
+
+from openassessment.xblock.utils.data_conversion import create_submission_dict
+from openassessment.xblock.apis.step_data_api import StepDataAPI
+
+
+class SelfAssessmentAPI(StepDataAPI):
```
Comment (jansenk) [inline openassessment/xblock/apis/assessments/self_assessment_api.py:8]: Are you not getting lint errors that these are missing docstrings?

Comment (nsprenkle) [reply openassessment/xblock/apis/assessments/self_assessment_api.py:8]: @jansenk , those are disabled by our `pylint` config... though we should probably have them, right @muselesscreator?

**Review by jansenk — APPROVED** [review]
This is such a massive PR, i really wish we could break it up somehow. That said, i can't find anyhting particularly objectionable herre

**Comment by nsprenkle** [comment]
> This is such a massive PR, i really wish we could break it up somehow. That said, i can't find anyhting particularly objectionable herre

I agree. I also got overruled 🙃

**Comment by nedbat** [comment]
CLA check, please.

**Comment by nedbat** [comment]
CLA check.

**Comment by nsprenkle** [comment]
Closing and reopening to re-run jobs

**Comment by nsprenkle** [comment]
Replacing with new PR to fix stuck job
